flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/AbstractFlinkService.java

```diff
@@ -160,19 +163,31 @@ public KubernetesClient getKubernetesClient() {
         return kubernetesClient;
     }
 
+    public CuratorFrameworkWithUnhandledErrorListener getCurator(Configuration conf) {
+        return ZooKeeperUtils.startCuratorFramework(conf, exception -> {});
+    }
+
     @Override
     public void submitApplicationCluster(
             JobSpec jobSpec, Configuration conf, boolean requireHaMetadata) throws Exception {
         LOG.info(
                 "Deploying application cluster{}",
                 requireHaMetadata ? " requiring last-state from HA metadata" : "");
+
+        // If Kubernetes or Zookeeper HA are activated, delete the job graph in HA storage so that
+        // the newly changed job config (e.g. parallelism) could take effect
         if (FlinkUtils.isKubernetesHAActivated(conf)) {
             final String clusterId = conf.get(KubernetesConfigOptions.CLUSTER_ID);
             final String namespace = conf.get(KubernetesConfigOptions.NAMESPACE);
-            // Delete the job graph in the HA ConfigMaps so that the newly changed job config(e.g.
-            // parallelism) could take effect
             FlinkUtils.deleteJobGraphInKubernetesHA(clusterId, namespace, kubernetesClient);
+        } else if (FlinkUtils.isZookeeperHAActivated(conf)) {
+            try (var curator = getCurator(conf)) {
```

Comment:
Here, the ZK client is created only for one request and then torn down (in `FlinkUtils ::isZookeeperHaMetadataAvailable` it's the same thing). We could somehow cache these clients but I think it's not worth it. 

First of all, these ZK requests happen quite rarely and the overhead of starting one client per request is negligible IMO.  

Moreover, if we decide to cache the client, we'll need to cache one per Flink application. Since we use the application `conf` when instantiating the client, we would also need to re-create the client every time ZK configuration values change in the application `conf`, which adds even more unnecessary complexity.

Reply:
I think it's fine to create the client on demand